### PR TITLE
3626: Simplify resource cache and fix first load

### PR DESCRIPTION
**Pull request by steffenkleinle**
### Short Description

This PR simplifies the resource cache and its usage according to the issue (#3626).
It also fixes an issue that the resource cache is not cached correctly on first app load.

### Proposed Changes

- Correctly cache resource cache on first load
- Simplify resource cache structure by omitting the grouping by page and using strings instead of object values
- Avoid passing resource cache to children and use it directly where needed

### Side Effects

N/A

### Testing

No testing necessary here, test #3522 very carefully instead.

### Resolved Issues

Fixes: #3626

---

### CODE REVIEW

File: native/src/utils/DefaultDataContainer.ts [hunk 1]
```diff
@@ -179,22 +141,12 @@ class DefaultDataContainer implements DataContainer {
     resourceCache: LanguageResourceCacheStateType,
   ): Promise<void> => {
     const context = new DatabaseContext(city)
-    const cache: Cache<CityResourceCacheStateType> = this.caches.resourceCache
-    const previousResourceCache = cache.getCached(context)
-
-    if (!previousResourceCache) {
-      await cache.cache(
```
Comment (steffenkleinle) [inline native/src/utils/DefaultDataContainer.ts]: This was the culprit @bahaaTuffaha of the cache not correctly updating on first load. We were setting the resource cache on the wrong cache instance since this is a local copy and not `this.caches.resourceCache` thats later read again.

File: native/src/components/RemoteContent.tsx [hunk 2]
```diff
@@ -73,6 +73,14 @@ const RemoteContent = ({
...
   const { width: deviceWidth } = useWindowDimensions()
 
+  const resourceMap: { [url: string]: string } = Object.entries(resourceCache).reduce(
+    (previous, [url, { filePath }]) => ({
+      ...previous,
+      [url]: getStaticServerFileUrl(filePath, staticServerUrl),
+    }),
+    {},
+  )
+
```
Comment (bahaaTuffaha) [inline native/src/components/RemoteContent.tsx]: I think it will create a new object for `...previous` each on each iteration.

Grok said it's complexity O(n²) so it gave me this as alternative (I couldn't comprehend the solution 😅 but looks clean and O(n)):
```js
const resourceMap = Object.fromEntries(
  Object.entries(resourceCache).map(([url, { filePath }]) => [
    url,
    getStaticServerFileUrl(filePath, staticServerUrl)
  ])
)
```

Comment (steffenkleinle) [reply native/src/components/RemoteContent.tsx]: Good suggestion, but this actually made me think and realize that we don't even need any object values and just `{ url: path }` is enough, therefore i further reduced the complexity and made this obsolete.

File: native/src/utils/__tests__/helpers.spec.ts
Comment (bahaaTuffaha) [inline native/src/utils/__tests__/helpers.spec.ts]: I think we can add a test for `getCachedResource` in case if the cache is empty to expect the url to be the same.

**Review by bahaaTuffaha — APPROVED** [review]
Nice work.. Tested on Android API 36 ✅️
